### PR TITLE
feat: (macOS) Support for Shapes

### LINKED ISSUE #2961: [macOS] Support for Shapes
## What would you like to be added:
Add support for `Windows.UI.Xaml.Shape` for macOS.

This also include support for the items below:

- Ellipse
- Line
- Path
- Polygon
- Polyline
- Rectangle

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): #
closes #2961 

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
Shapes are not supported in macOS

## What is the new behavior?
Shapes are supported in macOS

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Shapes/Ellipse.iOSmacOS.cs [hunk 1]
```diff
@@ -9,7 +9,9 @@ public partial class Ellipse : Shape
 	{
 		public Ellipse()
 		{
+#if __IOS__
 			ClipsToBounds = true;
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Shapes/Ellipse.iOSmacOS.cs:13]: is there and equivalent for macOS ?

File: src/Uno.UI/Media/PathStreamGeometryContext.cs [hunk 2]
```diff
@@ -37,6 +38,8 @@ public override void BeginFigure(Point startPoint, bool isFilled, bool isClosed)
 		{
 #if XAMARIN_IOS_UNIFIED || XAMARIN_IOS
 			bezierPath.MoveTo(startPoint);
+#elif __MACOS__
+			bezierPath.MoveTo(startPoint);
```
Comment (dr1rrb) [inline src/Uno.UI/Media/PathStreamGeometryContext.cs:42]:
```suggestion
#if XAMARIN_IOS_UNIFIED || XAMARIN_IOS || __MACOS__
			bezierPath.MoveTo(startPoint);
```

File: src/Uno.UI/UI/Xaml/Shapes/Rectangle.iOSmacOS.cs [hunk 3]
```diff
@@ -39,7 +41,7 @@ protected override Size ArrangeOverride(Size finalSize)
 #if __IOS__
 					? _BezierPath.FromRoundedRect(renderingArea, UIRectCorner.AllCorners, new CGSize(RadiusX, RadiusY)).CGPath
 #else
-					? _BezierPath.FromRoundedRect(area, (nfloat)RadiusX, (nfloat)RadiusY).ToCGPath()
+					? _BezierPath.FromRoundedRect(renderingArea, (nfloat)RadiusX, (nfloat)RadiusY).ToCGPath()
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Shapes/Rectangle.iOSmacOS.cs:44]: Oups! 🙃

File: src/Uno.UI/UI/Xaml/Shapes/Ellipse.iOSmacOS.cs (hunk 1 above)
Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Shapes/Ellipse.iOSmacOS.cs:13]: Yeah as @dr1rrb  said the `ClipToBounds=true` is the default behavior for macOS. When needing the opposite we will have to come up with workarounds. Thankfully for the Shapes this was not necessary.

**Pull request merged**